q2_types/feature_table/_type.py

```diff
@@ -30,5 +30,10 @@
 
 plugin.register_semantic_type_to_format(
     FeatureTable[Frequency | RelativeFrequency | PresenceAbsence],
-    artifact_format=FeatureTableDirectoryFormat
+    artifact_format=BIOMV210DirFmt
+)
+
+plugin.register_semantic_type_to_format(
```

Comment:
Ok, it looks like the behavior is "well-defined", so it's fine for the moment. @thermokarst mentioned that the tests where the reason it was added back. Ideally the framework will prevent these registrations where possible.

What's happening is the `type_format` lookup is actually a list, and a scan is done to find the subtypes (needed to make predicates work), for most uses, it will short-circuit asap. But the tests must iterate over all of them, which is why the second registration "worked".